plugins/storage/rdbms/events.go

```diff
@@ -232,24 +221,24 @@ func (r *RDBMS) FlushTestEvents() error {
 		}
 	}
 	r.buffTestEvents = nil
+
 	return nil
 }
 
 // GetTestEvents retrieves test events matching the query fields provided
 func (r *RDBMS) GetTestEvents(eventQuery *testevent.Query) ([]testevent.Event, error) {
 
-	if err := r.init(); err != nil {
-		return nil, fmt.Errorf("could not initialize database: %v", err)
-	}
-
 	// Flush pending events before Get operations
-	err := r.FlushTestEvents()
+	r.testEventsLock.Lock()
```

Comment:
Yes, share by communicating is the preferred golang way of doing things, and I am aware that this is not very idiomatic and should be reworked. The buffering was introduced in conjunction with prepared statements, and buffering the events seemed to be the right thing to do to make best used of a prepared statement (which was anyway removed later because not supported internally). However, then I realized that having a sync operation for some events is critical (who would have said..), so not all events should be buffered. At this point, unless we measure performance issue, as you say, I'll remove the buffering altogether.

Reply:
> buffering the events seemed to be the right thing to do to make best used of a prepared statement

In my opinion the way it was implemented it only adds fragileness and complexity here without solving any expected problem. If it would be implemented through channels then it would be more robust and simple (and even more performant), so at least it wouldn't be a problem itself. I guess we should leave it as is for now and just create a low-priority task after merging :)

Reply:
Will do in a separate PR though.